src/labkey/dom/Utils.ts

```diff
@@ -174,4 +175,121 @@ export function signalWebDriverTest(signalName: string, signalResult?: any): voi
+ * or DecimalFormat to control how the value is formatted</li>
+ * <li><b>timeOnly:</b> for dates, whether the date part should be ignored and only the time value is important</li>
+ * <li><b>forceString:</b> force the value to be treated as a string (i.e. prevent attempt to convert it to a date)</li>
+ * </ul>
+ * @example &lt;script type="text/javascript"&gt;
+ LABKEY.Utils.convertToExcel(
+ {
+         fileName: 'output.xls',
+         sheets:
+         [
+             {
+                 name: 'FirstSheet',
+                 data:
+                 [
+                     ['Row1Col1', 'Row1Col2'],
+                     ['Row2Col1', 'Row2Col2']
+                 ]
+             },
+             {
+                 name: 'SecondSheet',
+                 data:
+                 [
+                     ['Col1Header', 'Col2Header'],
+                     [{value: 1000.5, formatString: '0,000.00'}, {value: '5 Mar 2009 05:14:17', formatString: 'yyyy MMM dd'}],
+                     [{value: 2000.6, formatString: '0,000.00'}, {value: '6 Mar 2009 07:17:10', formatString: 'yyyy MMM dd'}]
+
+                 ]
+             }
+         ]
+     });
+ &lt;/script&gt;
+ */
+export const convertToExcel = (workbook: Workbook): void => {
+    const formData = { 'json': JSON.stringify(workbook) };
+    submitForm(buildURL("experiment", "convertArraysToExcel"), formData);
+};
+
+/**
+ * Sends a JSON object to the server which turns it into an TSV or CSV file and returns it to the browser to be saved or opened.
+ * @memberOf LABKEY.Utils
+ * @function
+ * @static
+ * @name convertToTable
+ * @param {Object} config.  The config object
+ * @param {String} config.fileNamePrefix name to suggest to the browser for saving the file. The appropriate extension (either ".txt" or ".csv", will be appended based on the delim character used (see below).  Defaults to 'Export'
+ * @param {String} config.delim The separator between fields.  Allowable values are 'COMMA' or 'TAB'.
+ * @param {String} config.quoteChar The character that will be used to quote each field.  Allowable values are 'DOUBLE' (ie. double-quote character), 'SINGLE' (ie. single-quote character) or 'NONE' (ie. no character used).  Defaults to none.
+ * @param {String} config.newlineChar The character that will be used to separate each line.  Defaults to '\n'
+ * @param {String} config.rows array of rows, which are arrays with values for each cell.
+ * @example &lt;script type="text/javascript"&gt;
+ LABKEY.Utils.convertToTable({
+         fileNamePrefix: 'output',
+         rows: [
+             ['Row1Col1', 'Row1Col2'],
+             ['Row2Col1', 'Row2Col2']
+         ],
+         delim: 'COMMA'
+     });
+ &lt;/script&gt;
+ */
+export const convertToTable = (config: TextTableForm): void => {
+    const formData = { 'json': JSON.stringify(config) };
+    submitForm(buildURL("experiment", "convertArraysToTable"), formData);
+};
+
+export interface Worksheet {
```

Comment:
nit: Consider moving these interfaces to the top of the file so they are declared before they are used.